### PR TITLE
Adds tabulate

### PR DESCRIPTION
**Pull request by marimeireles**
@wolfv please let me know if you want a colorful output somehow, I just made a basic thing.
I could maybe add a background for the row with the column names something like [this](https://github.com/p-ranav/tabulate#cell-colors).
I remember you saying something about not being able to use colors for some reason.

Anyway, if you have any input on how I should change the aesthetics, just let me know =)

### CODE REVIEW

**Comment by davidbrochart** [comment]
@marimeireles looks like you forgot to install tabulate in GH actions.

**Comment by marimeireles** [comment]
@davidbrochart for that I just have to add it on `conda install` in the ci?

**Comment by wolfv** [comment]
I think it would be good to hide the lines between the rows and maybe make the font in the header bold.

What do you think?

```
+----------------+---------+----------------+----------------------+
| Name           | Version | Build          | Channel              |
+----------------+---------+----------------+----------------------+
| jupyter_client | 6.1.2   | py_0           | installed            |
| jupyter_client | 4.2.2   | py27_0         | conda-forge/linux-64 |
| jupyter_client | 4.2.2   | py34_0         | conda-forge/linux-64 |
+----------------+---------+----------------+----------------------+
```


Also for the next level we should sort by version number and maybe merge the cells that have a common channel ... but we can do that later! :)

**Comment by davidbrochart** [comment]
@marimeireles Yes, is there a tabulate package on conda-forge? I looked quickly and I saw it was a Python package so I guess it's not the right one.

**Comment by wolfv** [comment]
`cpp-tabulate` should be the right one :)

File: include/query.hpp [hunk 1]
```diff
@@ -106,11 +113,14 @@ namespace mamba
             {
                 out << "No entries matching \"" << query << "\" found";
             }
+            tabulate::Table whatrequires_table_results;
+            // whatrequires_table_results({"Name", "Version", "Build", "Channel"});
```
[the remark this answers is not in the document]
Comment (wolfv) [reply include/query.hpp:117]: yep, whatrequires should work by using this syntax:

```
mamba --whatrequires jupyter_client

# or

mamba --whatrequires --installed jupyter_client
```

**Comment by marimeireles** [comment]
@wolfv Looks good. Good ideas =)
Did you see my review?

Thanks @davidbrochart and Wolf =)

**Comment by wolfv** [comment]
> @wolfv Looks good. Good ideas =)
> Did you see my review?

Yep already replied up there :)

**Comment by wolfv** [comment]
One other thought is that we will sooner or later need to have JSON output for these queries anyways. So we could also "just" figure out how to display a nice table from a nlohmann::json object then we have one format for everything.

the JSON could look something like 

```
{
packages: [
{name: blabla, version: 1.12.2, ...},
{name: blabla, version: 3.33, ...}
]
}
```

File: include/query.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (marimeireles) [reply include/query.hpp:126]: Using this command only returns:

```
~/Development/mamba(master*) » mamba --whatrequires jupyter_client

CommandNotFoundError: No command 'conda jupyter_client'.

```

This for me. I tried different packages.

**Comment by marimeireles** [comment]
Screenshot_2020-04-21_08-26-10
This is the new format.

**Comment by marimeireles** [comment]
Will we have to have a JSON output?
Or we already have?

**Comment by wolfv** [comment]
no we don't really have a JSON output yet, so let's keep the implementation as it is right now. 
I think @davidbrochart might have some initial work on this?

Do you want to rebase on master? Johan did a big refactor last night. The table looks good!

File: include/query.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wolfv) [reply include/query.hpp:126]: sorry, my fault. It's got to be `mamba repoquery --whatrequires jupyter_client` :bow:

**Comment by marimeireles** [comment]
Yeah, I'll do that.
Lemme just adjust the table for the `whatrequires` thing =)

**Comment by marimeireles** [comment]
Regarding ordering the rows by version:
I didn't find the option of fetching it ordered in the `solv` lib that we're using. I think the ordination that this lib give us is lexicographical.
So we have two choices here, either make a PR for `solv` offering them to do the right thing and order it numerically, or we do it locally. What do you think?

**Comment by wolfv** [comment]
I think we can skip the sorting for now as well, let's get the table in first :)

There is an option in tabulate: hide_border_XXX(), did you see it? I think it will be more condensed with that option:

https://github.com/p-ranav/tabulate/blob/f910903dfdabe04e428f3c7be64e426fdc2eb583/include/tabulate/format.hpp#L180-L226

**Comment by marimeireles** [comment]
Hmm sorry, didn't see that.
It's a better option indeed. Gonna change that. Thanks Wolf.

**Comment by marimeireles** [comment]
Hm! I don't get why the CI is failing though, on downloading `cpp-tabulate`.

**Comment by wolfv** [comment]
hey, we are using github actins now (it's in `.github/actions.../` if you add cpp-tabulate there it shoudl work fine. 

We can remove the travis file.

**Comment by marimeireles** [comment]
Want me to remove it now or later in a new PR?

**Comment by marimeireles** [comment]
Just tested the method you proposed and it doesn't work because it won't let me overwrite what it does later on (and I need to).

**Comment by marimeireles** [comment]
Weird errors on Mac. Gonna test them now.

**Comment by marimeireles** [comment]
@wolfv does it look good to you?
It runs on my macos :) (this was failing before the new tabulate version).
I wanted to make some changes, but I realized the API doesn't do what I thought it would. So I didn't really change anything.
Lemme know if you're interested in a more linear commit history. I can rebase it.

**Comment by wolfv** [comment]
awesome! yep, rebasing would be :+1: then we merge :)

**Pull request merged**